### DISCUSSION TITLE
Concerns about sensor attacks

### DISCUSSION THREAD

**Discussion by Mary0403** [paparazzi/paparazzi#3042] [General]
Dear everyone,

I have come across two sensor attacks that have been recently discussed in academia. Unlike most others, these specific attack vectors demand my serious attention. I am particularly interested to know if the Paparazzi software is vulnerable to these sensor attacks. If it is, I would like to understand how we can prevent such attacks at the control software level as well as hardware measures. Details are below.

1. Intentional sound noise causes resonant on IMUs
[Motivating Example - YouTube](https://youtu.be/TXStXvyV3To)
Details: [UnRocker](https://sites.google.com/view/unrocker/)

2. Electromagnetic interference (EMI) disturbs sensor readings on SPI/I2C buses
[EM injection experiments on a hovering drone - YouTube](https://www.youtube.com/watch?v=7LN8quYFThI)
Details: [Paralyzing Drones](https://sites.google.com/view/paralyzing-drones-via-emi)

These two attacks potentially make the drone crash on the ground, as shown in the demo videos. Thus, I would greatly appreciate your insights and opinions on these sensor attacks. Thank you!

**Comment by gautierhattenberger** [comment]
I moved to discussion since this is not an actual software issue.

There are no particular actions taken in Paparazzi to counter this kind of "hardware" attack. Some studies have been done in my lab to detect spoofing attacks on a the communication network, not sure it would apply so well in the case of IMU jamming. If you loose the IMU data, there is not much you can do anyway.
Hardware protection against EMI or even sound attacks (I never heard of this before) is a something for board designer and integrators, not much related to the software part I think.
